runtime/vm/options.c

```diff
@@ -0,0 +1,198 @@
+ */
+
+
+/* Option types (first ten entries for compatibility) */
+#define TYPE_BOOL 0
+#define TYPE_CCSTR 1
+#define TYPE_CCSTRLIST 2
+#define TYPE_DOUBLE 3
+#define TYPE_INT 4
+#define TYPE_INTX 5
+#define TYPE_SIZET 6
+#define TYPE_UINT 7
+#define TYPE_UINT64T 8
+#define TYPE_UINTX 9
+
+static const char *option_type_string[] = {
+	"bool",
+	"ccstr",
+	"ccstrlist",
+	"double",
+	"int",
+	"intx",
+	"size_t",
+	"uint",
+	"uint64_t",
+	"uintx"
+};
+#define NUM_TYPE_STRINGS	(sizeof(option_type_string) / sizeof(option_type_string[0]))
+
+typedef union J9OptionValue {
+	UDATA v_uintx;
+	U_64 v_uint64;
+	size_t v_sizet;
+	double v_double;
+	char *v_ccstr;
+	char **v_ccstrlist;
+	U_32 v_uint;
+	IDATA v_intx;
+	I_8 v_bool;
+} J9OptionValue;
+
+#define CAT_PRODUCT		0
+#define CAT_ARCHPRODUCT		1
+#define CAT_PDPRODUCT		2
+#define CAT_PRODUCTRW		3
+#define CAT_DIAGNOSTIC		4
+#define CAT_EXPERIMENTAL	5
+#define CAT_MANAGEABLE		6
+
+static const char *option_category_string[] = {
+	"{product}",
+	"{ARCH product}",
+	"{pd product}",
+	"{product rw}",
+	"{diagnostic}",
+	"{experimental}",
+	"{manageable}",
+};
+#define NUM_CATEGORY_STRINGS	(sizeof(option_category_string) / sizeof(option_category_string[0]))
+
+#define SET_DEFAULT	0
+#define SET_COMMANDLINE	1
+#define SET_ERGONOMIC	2
+
+static const char *option_howset_string[] = {
+	"{default}",
+	"{command line}",
+	"{ergonomic}"
+};
+#define NUM_HOWSET_STRINGS	(sizeof(option_howset_string) / sizeof(option_howset_string[0]))
+
+typedef struct J9OptionDesc {
+	const char * name;
+	I_8 type;
+	I_8 category;
+	I_8 howset;
+	J9OptionValue value;
+} J9OptionDesc;
+
+
+/* options_enum gives each option an index starting from zero */
+typedef enum options_enum {
+	NoOption=-1,
+
+	#define OPTION(n,t,cats,how) IDX_ ## n,
+	#include "options.inc"
+	#undef OPTION
+} options_enum;
+
+/* options array lists all options, same order as options_enum */
+static J9OptionDesc options[]={
+	#define OPTION(n,t,cats,how) { n, t, cats, how, { 0 } },
+	#include "options.inc"
+	#undef OPTION
+};
+#define NUM_OPTIONS	(sizeof(options)/sizeof(options[0]))
+
+
+/*
+ * initializeFlagSettings initializes the value and howset fields for all options in
+ *	the options array. Constant default values would get howset=SET_DEFAULT, values
+ *	that are computed heuristically based on other options or the environment get
+ *	howset=SET_ERGONOMIC, and if the value was set via a command line option (including
+ *	aliases like -XX::MaxMemorySize= and -Xmx) get howset=SET_COMMANDLINE .
+ */
+void initializeFlagSettings(J9JavaVM* vm) {
+	PORT_ACCESS_FROM_JAVAVM(vm);
+	J9OptionDesc *option=NULL;
+
+	/* Initialize value for -XX:MaxHeapSize / -Xmx */
+	option = options + IDX_VMOPT_MAXHEAPSIZE;
+	option->value.v_sizet = vm->memoryManagerFunctions->j9gc_get_maximum_heap_size(vm);
+	if ((findArgInVMArgs( PORTLIB, vm->vmArgsArray, STARTSWITH_MATCH, VMOPT_XMX, NULL, 0) >= 0))
```

Comment:
please always use `{` & `}` for control flow.  It's too easy to add a debug printf into 1 line if statement and change the meaning of the program.  By always using curly braces, we can avoid that entire class of issues